plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestStandardColumnMappings.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+
+import static io.trino.plugin.jdbc.StandardColumnMappings.verifyLongTimestamp;
+import static io.trino.plugin.jdbc.StandardColumnMappings.verifyShortTimestamp;
+import static io.trino.spi.type.Timestamps.MICROSECONDS_PER_SECOND;
+import static io.trino.spi.type.Timestamps.NANOSECONDS_PER_MICROSECOND;
+import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_NANOSECOND;
+import static java.lang.Math.floorDiv;
+import static java.time.ZoneOffset.UTC;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestStandardColumnMappings
+{
+    @Test(dataProvider = "validShortTimestamps")
+    public void testVerifyValidShortTimestamps(int precision, LocalDateTime localDateTime, int picosOfMicro)
+    {
+        Instant instant = localDateTime.atZone(UTC).toInstant();
+        long epochMicros = instant.getEpochSecond() * MICROSECONDS_PER_SECOND + floorDiv(instant.getNano(), NANOSECONDS_PER_MICROSECOND);
+        picosOfMicro += (instant.getNano() % NANOSECONDS_PER_MICROSECOND) * PICOSECONDS_PER_NANOSECOND;
+        verifyShortTimestamp(epochMicros, picosOfMicro, precision);
+    }
+
+    @DataProvider
+    public Object[][] validShortTimestamps()
+    {
+        return new Object[][] {
+                {0, LocalDateTime.of(2018, 3, 25, 3, 17, 17), 0},
```

Comment:
I like data provides, but i found there is typically a more readable way to structure the tests:

```
testFoo() {
 testFoo(1);
 testFoo(2);
 testFoo(3);
}
```

it gets all the power of java completion, it works nice in the IDE (failure clicks into the failing line)

it does _not_ have the advantage that the data point is part of test label, but for such long labels, it's often not readable anyway

Reply:
Refactor, looks _much_ better, thanks!